python/ray/util/spark/cluster_init.py

```diff
@@ -210,6 +229,31 @@ def shutdown(self, cancel_background_job=True):
                 )
             self.is_shutdown = True
 
+    def to_dict(self):
+        ignore_fields = {
+            "head_proc",
+            "start_hook",
+            "is_shutdown",
+            "spark_job_is_canceled",
+            "background_job_exception",
+        }
+        cluster_info = {
+            str(k): v for k, v in self.__dict__.items() if k not in ignore_fields
+        }
+        cluster_info["head_proc_pid"] = self.head_proc.pid
+        return cluster_info
+
+    @classmethod
+    def from_dict(cls, cluster_info):
+        head_proc_pid = cluster_info.pop("head_proc_pid", None)
```

Comment:
Risk: the head node process might died but then another irrelevent 
 process started and reusing the pid, then you might kill irrelevent process when calling `shutdown_ray_cluster`. 

Can we get ray head process pid from ray head port ?